src/main/java/com/ghgande/j2mod/modbus/io/ModbusSerialTransport.java

```diff
@@ -591,19 +601,68 @@ void waitBetweenFrames(int transDelayMS, long lastTransactionTimestamp) {
             // Make use we have a gap of 3.5 characters between adjacent requests
             // We have to do the calculations here because it is possible that the caller may have changed
             // the connection characteristics if they provided the connection instance
-            int delay = (int) (Modbus.INTER_MESSAGE_GAP * (commPort.getNumDataBits() + commPort.getNumStopBits()) * 1000 / commPort.getBaudRate());
-
-            // If the delay is below the miimum, set it to the minimum
-            if (delay > Modbus.MINIMUM_TRANSMIT_DELAY) {
-                delay = Modbus.MINIMUM_TRANSMIT_DELAY;
-            }
-
+            int delay = getInterframeDelay() / 1000;
+            
             // How long since the last message we received
-            long gapSinceLastMessage = System.currentTimeMillis() - lastTransactionTimestamp;
+            long gapSinceLastMessage = (System.nanoTime() - lastTransactionTimestamp) / NS_IN_A_MS;
             if (delay > gapSinceLastMessage) {
-                ModbusUtil.sleep(delay - gapSinceLastMessage);
+                long sleepTime = delay - gapSinceLastMessage;
+                
+                ModbusUtil.sleep(sleepTime);
+                
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Waited between frames for {} ms", sleepTime);
+                }
             }
         }
     }
-
+    
+    /**
+     * In microseconds
+     * @return
+     */
+    int getInterframeDelay() {
+        if (commPort.getBaudRate() > 19200) {
+            return 1750;
+        } else {
+            return Math.max(getCharInterval(Modbus.INTER_MESSAGE_GAP), Modbus.MINIMUM_TRANSMIT_DELAY);
+        }
+    }
+    
+    /**
+     * The maximum delay between characters in microseconds
+     * @return
+     */
+    long getMaxCharDelay() {
+        if (commPort.getBaudRate() > 19200) {
+            return 1750;
```

Comment:
This should be 750 from the modbus specification but that did not seem to work in practice, probably OS delays and other factors. Should this delay be configurable... ? Seems kind of overkill